packages/web/src/pages/settings-page/components/desktop/DeveloperApps/EditAppPage.tsx

```diff
@@ -0,0 +1,211 @@
+import { useCallback, useEffect, useState } from 'react'
+
+import { IconCopy, IconButton, Button, Flex, IconEmbed } from '@audius/harmony'
+import { Divider } from 'components/divider'
+import Toast from 'components/toast/Toast'
+import { MountPlacement } from 'components/types'
+import { copyToClipboard } from 'utils/clipboardUtil'
+import styles from './EditAppPage.module.css'
+import { CreateAppPageProps, CreateAppsPages } from './types'
+import { Name, Status } from '@audius/common/models'
+import { accountSelectors } from '@audius/common/store'
+import { Form, Formik, useField } from 'formik'
+import { z } from 'zod'
+import { toFormikValidationSchema } from 'zod-formik-adapter'
+
+import { make, useRecord } from 'common/store/analytics/actions'
+import { TextAreaField, TextField } from 'components/form-fields'
+import { useSelector } from 'utils/reducer'
+import { DEVELOPER_APP_DESCRIPTION_MAX_LENGTH, DEVELOPER_APP_IMAGE_URL_MAX_LENGTH, DEVELOPER_APP_NAME_MAX_LENGTH, developerAppEditSchema, useEditDeveloperApp } from '@audius/common/api'
+
+const { getUserId } = accountSelectors
+
+type EditAppPageProps = CreateAppPageProps
+
+type DeveloperAppValues = z.input<typeof developerAppEditSchema>
+
+const messages = {
+  appNameLabel: 'App Name',
+  imageUrlLabel: 'App Icon URL',
+  apiKey: 'api key',
+  copyApiKeyLabel: 'copy api key',
+  copied: 'Copied!',
+  goBack: 'Back to Your Apps',
+  back: 'Back',
+  save: 'Save Changes',
+  saving: 'Saving',
+  miscError: 'Sorry, something went wrong. Please try again later.'
+}
+
+const ImageField = ({ name }: { name: string }) => {
+  const [{ value }] = useField(name)
+  return value
+    ? <img src={value} height='136' width='136' />
```

Comment:
Assuming the hard-coded sizes in this component are a case of not-yet-harmonized design?

Reply:
probably won't have something like this harmonized for a while unf, but yeah, it's a good point!